benchmarks/queries/q8.sql

```diff
@@ -1,9 +1,9 @@
 select
     o_year,
-    cast(cast(sum(case
-                      when nation = 'BRAZIL' then volume
-                      else 0
-        end) as decimal(12,2)) / cast(sum(volume) as decimal(12,2)) as decimal(15,2)) as mkt_share
+    sum(case
+            when nation = 'BRAZIL' then volume
+            else 0
+        end) / sum(volume) as mkt_share
```

Comment:
These cast are added before to work around overflow issue.

After https://github.com/apache/arrow-datafusion/pull/6832, q8 should pass without cast.

Reply:
Looks okay, all tests are passed.